zookeeper-server/src/main/java/org/apache/zookeeper/server/FinalRequestProcessor.java

```diff
@@ -468,6 +470,28 @@ public void processRequest(Request request) {
                 rsp = new GetChildren2Response(children, stat);
                 break;
             }
+            case OpCode.getChildrenList: {
+                lastOp = "GETC";
+                GetChildrenListRequest getChildrenListRequest = new GetChildrenListRequest();
+                ByteBufferInputStream.byteBuffer2Record(request.request,
+                        getChildrenListRequest);
+                List<List<String>> childrenList = new ArrayList<List<String>>();
+                for(String p : getChildrenListRequest.getPathList()) {
+                    path = p;
+                    DataNode n = zks.getZKDatabase().getNode(path);
+                    if (n == null) {
```

Comment:
Not sure if it makes sense to return empty list if the node doesn't exist, so that we won't waste the cycle to fail often due to nodes being created and deleted often.

Reply:
Yeah, I was considering something like that, however, this way we could not differentiate whether the request failed or it is a node without children.